server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -317,18 +326,57 @@ public IndexInput openInput(String name, IOContext context) throws IOException {
         }
     }
 
-    public void copyFrom(Directory from, String src, String dest, IOContext context, boolean useCommonSuffix) throws IOException {
+    @Override
+    public void acquireLock(long primaryTerm, long generation, String resourceID, String ttl) throws IOException {
+        String metadataFile = getMetadataFileForCommit(primaryTerm, generation);
+
+        mdLockManager.acquire(ShardLockInfo.getLockInfoBuilder().withFileToLock(metadataFile)
+            .withResourceId(resourceID).withTTL(ttl).build());
+
+        // Confirming that the metadata file for which we acquired lock still persist in remote store.
+        if (!metadataFile.equals(getMetadataFileForCommit(primaryTerm, generation))) {
+            mdLockManager.release(ShardLockInfo.getLockInfoBuilder().withResourceId(resourceID).build());
```

Comment:
what is the behavior if this `release` fails ? seems like it will leak the lock file ?

Reply:
removed this call.